helical/models/classification/neural_network.py

```diff
@@ -2,18 +2,25 @@
 from typing_extensions import Self
 from sklearn.preprocessing import LabelEncoder
 import numpy as np
-import tensorflow as tf
-from tensorflow.keras.models import Sequential
-from tensorflow.keras.layers import Dense, Dropout
-from tensorflow.keras.optimizers import Adam
-from tensorflow.keras.metrics import F1Score
-from tensorflow.keras.utils import to_categorical
 from helical.models.base_models import BaseTaskModel
 from pathlib import Path
+import torch
+import torch.nn as nn
+import torch.optim as optim
+from torch.utils.data import DataLoader, TensorDataset
+import logging
 
+LOGGER = logging.getLogger(__name__)
 class NeuralNetwork(BaseTaskModel):
-    def __init__(self, loss: str = "categorical_crossentropy", learning_rate: float = 0.001, epochs=10, batch_size=32) -> None:
-        self.loss = loss
+    def __init__(self, loss: str = "categorical_crossentropy",learning_rate: float = 0.001, epochs=10, batch_size=32) -> None:
```

Comment:
nit: we could probably rename "categorical_crossentropy" as just "crossentropy". The motivation for this is that categorical_crossentropy is a term used mostly in Keras. In Pytorch the CrossEntropyLoss works almost the same in the same way, but with a different name.

It's a very minor comment. Feel free to ignore.

Reply:
Maybe we can take the loss function in as a parameter instead of a string because this limits us to only the loss functions we have implemented?